identity-core/src/common/timestamp.rs

```diff
@@ -144,6 +147,27 @@ impl TryFrom<&'_ str> for Timestamp {
   }
 }
 
+// This struct is only used to (potentially) avoid an allocation when deserializing a Timestamp. We cannot deserialize
+// via &str because that breaks serde_json::from_value which we use extensively in the Stronghold bindings.
+// This approach is inspired by https://crates.io/crates/serde_str_helpers, but we only use a subset of the functionality offered by that crate.
+#[derive(Deserialize, Serialize)]
+struct ProvisionalTimestamp<'a>(#[serde(borrow)] Cow<'a, str>);
+
+impl<'a> Deref for ProvisionalTimestamp<'a> {
+  type Target = str;
+  fn deref(&self) -> &Self::Target {
+    self.0.borrow()
+  }
+}
+
+impl<'a> TryFrom<ProvisionalTimestamp<'a>> for Timestamp {
+  type Error = Error;
+
+  fn try_from(value: ProvisionalTimestamp<'a>) -> Result<Self, Self::Error> {
+    Timestamp::parse(&value)
+  }
+}
```

Comment:
Nit: only used in one place so I wouldn't implement `Deref`.

```suggestion
impl<'a> TryFrom<ProvisionalTimestamp<'a>> for Timestamp {
  type Error = Error;

  fn try_from(value: ProvisionalTimestamp<'a>) -> Result<Self, Self::Error> {
    Timestamp::parse(value.0.borrow())
  }
}
```